### PR TITLE
Added TooltipController to manage visibility of a tooltip

### PR DESCRIPTION
**Pull request by mhbdev**
*In response to #126787 feature request, In this PR a TooltipController can be passed to the Tooltip widget to manage its visibility and the user can `show` or `conceal` the tooltip using this controller.*

#126787

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

It is not a breaking change.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/material/tooltip.dart [hunk 1]
```diff
@@ -893,3 +926,55 @@ class _TooltipOverlay extends StatelessWidget {
...
+///   [TooltipController].
+class TooltipController {
+  /// The tooltip that this controller controls.
+  ///
+  /// This is set automatically when a [TooltipController] is given to the
+  /// tooltip it controls.
+  TooltipState? _state;
+
+  /// Whether or not the associated tooltip is currently open.
+  bool get isOpen {
```
Comment (LongCatIsLooong) [inline packages/flutter/lib/src/material/tooltip.dart:948]: When would the developer want to use `isOpen`?  Also if the tooltip is fading in/out I'm not sure what `isOpen` is supposed to indicate?

File: packages/flutter/lib/src/material/tooltip.dart [hunk 2]
```diff
@@ -341,6 +346,7 @@ class TooltipState extends State<Tooltip> with SingleTickerProviderStateMixin {
   static const bool _defaultEnableFeedback = true;
   static const TextAlign _defaultTextAlign = TextAlign.start;
 
+  TooltipController? _internalTooltipController;
```
Comment (LongCatIsLooong) [inline packages/flutter/lib/src/material/tooltip.dart:349]: If I'm understanding it correctly, a controller is for the developer to control a single tooltip and is never used internally by the `Tooltip` widget. If the developer never specified a controller then why do we need to create an internal one?

File: packages/flutter/lib/src/material/tooltip.dart (hunk 1 above)
Comment (mhbdev) [reply packages/flutter/lib/src/material/tooltip.dart:948]: > When would the developer want to use `isOpen`? Also if the tooltip is fading in/out I'm not sure what `isOpen` is supposed to indicate?

You are right. I was thinking of a situation in which user wants to open another tooltip when the other one is already opened/closed. But as you mentioned that tooltips are fading in/out there isOpen would be useless.

File: packages/flutter/lib/src/material/tooltip.dart (hunk 2 above)
Comment (mhbdev) [reply packages/flutter/lib/src/material/tooltip.dart:349]: I was using MenuAnchor approach to implement the controller for the tooltip. There may be further needs for this internal controller but you are totally right.

**Pull request closed without merging**